### PR TITLE
Add Vendors mutations

### PR DESCRIPTION
**Pull request by kewitz**
Solves https://github.com/opencollective/opencollective/issues/7047

### CODE REVIEW

File: server/graphql/v2/input/VendorInput.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    notes: { type: GraphQLString },
+  }),
+});
+
+export const GraphQLVendorCreateInput = new GraphQLInputObjectType({
+  name: 'VendorCreateInput',
+  fields: () => ({
+    name: { type: new GraphQLNonNull(GraphQLString) },
+    legalName: { type: GraphQLString },
+    tags: { type: new GraphQLList(GraphQLString) },
```
Comment (Betree) [inline server/graphql/v2/input/VendorInput.ts:37]: I like to use `GraphQLNonEmptyString` for these cases 

```suggestion
    name: { type: new GraphQLNonNull(GraphQLNonEmptyString) },
    legalName: { type: GraphQLString },
    tags: { type: new GraphQLList(GraphQLNonEmptyString) },
```

File: server/graphql/v2/mutation/VendorMutations.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        ParentCollectiveId: host.id,
+        ...pick(args.vendor, ['name', 'legalName', 'tags']),
+        data: {
+          vendorInfo: pick(vendorInfo, VENDOR_INFO_FIELDS),
+        },
+      };
+
+      if (vendorInfo.taxType) {
+        assert(vendorInfo.taxId, new BadRequest('taxId is required when taxType is provided'));
+        vendorData.data[vendorInfo.taxType] = vendorInfo.taxId;
```
Comment (Betree) [inline server/graphql/v2/mutation/VendorMutations.ts:57]: If we aim to re-use existing fields, that would be:

```suggestion
        vendorData.data[vendorInfo.taxType] = { number: vendorInfo.taxId };
```

File: server/graphql/v2/mutation/VendorMutations.ts
Comment (Betree) [inline server/graphql/v2/mutation/VendorMutations.ts:1]: We should make sure to check the scope (`enforceScope` or `checkScope`) for all the mutations in this file

File: server/graphql/v2/mutation/VendorMutations.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      },
+    },
+    resolve: async (_, args, req) => {
+      if (!req.remoteUser) {
+        throw new Unauthorized('You need to be authenticated to perform this action. Please login and try again.');
+      }
+
+      const vendor = await fetchAccountWithReference(args.vendor, { loaders: req.loaders, throwIfMissing: true });
+      const host = await req.loaders.Collective.byId.load(vendor.ParentCollectiveId);
+      assert(host, new NotFound('Vendor host not found'));
```
Comment (Betree) [inline server/graphql/v2/mutation/VendorMutations.ts:85]: We should check the account type

```suggestion
      assert(host, new NotFound('Vendor host not found'));
      assert(vendor.type === 'VENDOR', new ValidationFailed('Account is not a vendor'));
```

File: server/graphql/v2/mutation/VendorMutations.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        ...pick(args.vendor, ['name', 'legalName', 'tags']),
+        data: {
+          ...vendor.data,
+          vendorInfo: { ...vendor.data?.vendorInfo, ...pick(vendorInfo, VENDOR_INFO_FIELDS) },
+        },
+      };
+
+      if (vendorInfo.taxType) {
+        assert(vendorInfo.taxId, new BadRequest('taxId is required when taxType is provided'));
+        vendorData.data[vendorInfo.taxType] = vendorInfo.taxId;
```
Comment (Betree) [inline server/graphql/v2/mutation/VendorMutations.ts:102]:
```suggestion
        vendorData.data[vendorInfo.taxType] = { number: vendorInfo.taxId };
```

**Review by Betree — COMMENTED** [review]
A few checks to add but the core looks perfect :ok_hand:

File: server/graphql/v2/mutation/VendorMutations.ts (hunk 2 above)
Comment (kewitz) [reply server/graphql/v2/mutation/VendorMutations.ts:57]: @Betree should we presume `type: 'OWN'` as well?

Comment (Betree) [reply server/graphql/v2/mutation/VendorMutations.ts:57]: We can, yes

**Pull request merged**